server/controllers/userController.js

```diff
@@ -188,6 +189,34 @@ class UsersController {
     }
   }
 
+  static async updateUser(req, res) {
+    try {
+      const { email, first_name, last_name, dob, year_of_graduation, phone_number, role } = res.locals.user;
+      const loggedInUser = req.session.role;
+      const updateData = { first_name, last_name, dob, year_of_graduation, phone_number };
+      // only admin & super admin can change a user's role
+      if (['admin', 'super admin'].includes(loggedInUser)) {
+        updateData.role = role;
```

Comment:
Job well done. One question. Should a teacher attempts to update one user's role, what would the response be? Will the update operation breaks and alert the teacher with a descriptive message? I feel the operation continue as if nothing unusual happens. Or do I miss something? @jamsgra

Reply:
If a teacher attempts to update any user other than student they will either get: 

`Sorry, you do not have the required privilege to update admin or super admin`

or

`Sorry, you do not have the required privilege to update teacher`

So they can only update students. And if they try to update a student's role they will get:

`Sorry, you do not have the required privilege to update role`